internal/tests/reactable/bloom_test.go

```diff
@@ -126,5 +126,93 @@ func TestECBloom(t *testing.T) {
 	if trg[0].AuraContains(attributes.Hydro, attributes.Dendro) {
 		t.Errorf("expecting target to not contain any remaining hydro or dendro aura, got %v", trg[0].ActiveAuraString())
 	}
+}
+
+func TestBloomSeedLimit(t *testing.T) {
+	c, trg := makeCore(10)
+	err := c.Init()
+	if err != nil {
+		t.Errorf("error initializing core: %v", err)
+		t.FailNow()
+	}
+
+	c.QueueAttackEvent(&combat.AttackEvent{
+		Info: combat.AttackInfo{
+			Element:    attributes.Hydro,
+			Durability: 25,
+		},
+		Pattern: combat.NewCircleHit(trg[0], 100, false, combat.TargettableEnemy, combat.TargettableGadget),
+	}, 0)
+	advanceCoreFrame(c)
+	c.QueueAttackEvent(&combat.AttackEvent{
+		Info: combat.AttackInfo{
+			Element:    attributes.Dendro,
+			Durability: 25,
+		},
+		Pattern: combat.NewCircleHit(trg[0], 100, false, combat.TargettableEnemy, combat.TargettableGadget),
+	}, 0)
+	advanceCoreFrame(c)
+
+	for i := 0; i < reactable.DendroCoreDelay+1; i++ {
+		advanceCoreFrame(c)
+	}
+
+	if c.Combat.GadgetCount() != 5 {
+		t.Errorf("expected only 5 seeds remaining, got %v", c.Combat.GadgetCount())
+	}
+}
+
+func TestBloomOldestDeleted(t *testing.T) {
+	c, trg := makeCore(1)
+	err := c.Init()
+	if err != nil {
+		t.Errorf("error initializing core: %v", err)
+		t.FailNow()
+	}
+
+	//oldest should be the 2nd one, which is frame 3 ?
+	for i := 0; i < 6; i++ {
+		c.QueueAttackEvent(&combat.AttackEvent{
+			Info: combat.AttackInfo{
+				Element:    attributes.Hydro,
+				Durability: 25,
+			},
+			Pattern: combat.NewCircleHit(trg[0], 100, false, combat.TargettableEnemy, combat.TargettableGadget),
+		}, 0)
+		advanceCoreFrame(c)
+		c.QueueAttackEvent(&combat.AttackEvent{
+			Info: combat.AttackInfo{
+				Element:    attributes.Dendro,
+				Durability: 25,
+			},
+			Pattern: combat.NewCircleHit(trg[0], 100, false, combat.TargettableEnemy, combat.TargettableGadget),
+		}, 0)
+		advanceCoreFrame(c)
+	}
+
+	for i := 0; i < reactable.DendroCoreDelay+1; i++ {
+		advanceCoreFrame(c)
+	}
+
+	if c.Combat.GadgetCount() != 5 {
+		t.Errorf("expected only 5 seeds remaining, got %v", c.Combat.GadgetCount())
+	}
+
+	//find oldest
+	f := math.MaxInt64
+	oldest := -1
+	for i, v := range c.Combat.Gadgets() {
+		if v == nil || v.GadgetTyp() != combat.GadgetTypDendroCore {
+			continue
+		}
+		if v.Src() < f {
+			f = v.Src()
+			oldest = i
+		}
+	}
+	og := c.Combat.Gadget(oldest)
+	if og.Src() != 3+reactable.DendroCoreDelay {
```

Comment:
Consider: checking for the oldest bloom by `key` instead of `src`. The key should be `2` in this test.

Reply:
should be the same. src might be more reliable because each reaction is 2 frames apart. no guarantee that key will always be 2. if we do key should then store the key